lib/navigation/filtering/extended_kalman_filter.cpp

```diff
@@ -0,0 +1,42 @@
+#include "extended_kalman_filter.hpp"
+
+namespace hyped::navigation {
+
+ExtendedKalmanFilter::ExtendedKalmanFilter(std::shared_ptr<core::ITimeSource> time_source,
+                                           const StateVector initial_state,
+                                           const ErrorCovarianceMatrix initial_error_covariance)
+    : time_source_(time_source),
+      last_update_time_(time_source->now()),
+      state_estimate_(initial_state),
+      error_covariance_(initial_error_covariance)
+{
+  static_assert(state_dimension > 0);
```

Comment:
HYPED standards have normally been to use a create function which returns an optional<class> for error handling in the constructor (most of the sensors are a good example)

Reply:
this is fine if we are defining the `state_dimension` in a `constexpr` then this catches that at compile time right? we would only need a `create` if these are in a config file, which idk enough about kalman filter dimensions to say if we want to do that